cafebabel/articles/views.py

```diff
@@ -28,3 +43,152 @@ def article_proposal_create():
     mail.send(msg)
     flash('Your proposal was successfully send.', 'success')
     return redirect(url_for('home'))
+
+
+@draft_bp.route('/')
+@editor_required
+@login_required
+def draft_create_form():
+    article = Article()
+    authors = User.objects.all()
+    return render_template('articles/edit.html', article=article,
+                           authors=authors)
+
+
+@draft_bp.route('/', methods=['post'])
+@editor_required
+@login_required
+def draft_create():
+    article = Article()
+    data = request.form.to_dict()
+    data['editor'] = current_user.id
+    if data.get('author'):
+        data['author'] = User.objects.get(id=data.get('author'))
+    for field, value in data.items():
+        setattr(article, field, value)
+    image = request.files.get('image')
+    if image:
+        article.attach_image(image)
+    article.save()
+    flash('Your article was successfully saved.')
+    if article.is_draft:
+        return redirect(url_for('draft.draft_detail', draft_id=article.id))
+    else:
+        return redirect(url_for('article.article_detail', slug=article.slug,
+                                id=article.id))
+
+
+@draft_bp.route('/<regex("\w{24}"):draft_id>/edit/')
+def draft_edit_form(draft_id):
+    try:
+        article = Article.objects.get(id=draft_id)
+    except Article.DoesNotExist:
+        abort(404, 'No draft found with this id.')
+    authors = User.objects.all()
+    return render_template('articles/edit.html', article=article,
+                           authors=authors)
+
+
+@draft_bp.route('/<regex("\w{24}"):draft_id>/edit/',
+                methods=['post'])
+def draft_edit(draft_id=None):
+    try:
+        article = Article.objects.get(id=draft_id)
+    except Article.DoesNotExist:
+        abort(404, 'No draft found with this id.')
+    data = request.form.to_dict()
+
+    data['editor'] = current_user.id
+    if data.get('author'):
+        data['author'] = User.objects.get(id=data.get('author'))
+    for field, value in data.items():
+        setattr(article, field, value)
+    if data.get('delete-image'):
+        article.delete_image()
+    image = request.files.get('image')
+    if image:
+        article.attach_image(image)
+    article.save()
+    flash('Your article was successfully saved.')
+    if article.is_draft:
+        return redirect(url_for('draft.draft_detail', draft_id=article.id))
+    else:
+        return redirect(url_for('article.article_detail', slug=article.slug,
+                                article_id=article.id))
+
+
+@draft_bp.route('/<regex("\w{24}"):draft_id>/')
+def draft_detail(draft_id):
+    try:
+        article = Article.objects.get(id=draft_id)
```

Comment:
Ensure the status is _draft_.
Eventually redirect to the published article otherwise?